loadbalancing.md

```diff
@@ -0,0 +1,345 @@
+
+It is also possible to use DNS round-robin. However this is not advisable as this won't prevent connections from going to a machine that is unavailable. This means that your application may have to re-try connections more often if not all servers are available.
+
+Some of the newer MySQL connectors have support for [DNS SRV](https://dev.mysql.com/doc/refman/8.0/en/connecting-using-dns-srv.html). This is somewhat similar to round-robin DNS, but allows you to set priorities. The benefit from this is that it is an industry standard and that this makes the client-side configuration easier.
+
+If you are using Kubernetes you probably want to use the [LoadBalancer](https://docs.pingcap.com/tidb-in-kubernetes/stable/access-tidb#loadbalancer) service type that works with Amazon ELB and similar services from other cloud vendors. For on-premise Kubernetes something like [metallb](https://metallb.universe.tf/) can be used.
+
+When your are not using Kubernetes to deploy on a cloud service you can still use the loadbalancing service from your cloud provider.
+
+Another common option is to use a software based loadbalancer. Common ones are ProxySQL, haproxy and MySQL Router.
+
+The last type of loadbalancer that we will discuss here is hardware based loadbalancers. These are physical machines where you plug in a network cable. These are often costly, but also offer a high throughput.
+
+## General loadbalancer requirements for TiDB
+
+Where loadbalancers may offer advanced features like read/write splitting for MySQL, this is not needed for TiDB. The loadbalancer does not need to be able to inspect the MySQL protocol. A TCP level loadbalacer works fine for TiDB.
+
+The loadbalancer only needs access to the TiDB servers (TCP port 4000 by default) and doesn't need access to PD or TiKV servers.
+
+In general a TCP health check is sufficient. Health checks that check for `read_only` or other MySQL or InnoDB variables should be avoided.
+
+Another option is the HTTP status API that is by default available on `http://<ip_of_tidb>:10080/status`. The `graceful-wait-before-shutdown` variable can be used to configure tidb to wait for a number of seconds between starting to shutdown and really closing network connections. In this time the HTTP healthcheck will already fail allowing loadbalancers to start and redirect traffic elsewhere.
```

Comment:
I think we can remove this sentence if we use the suggestion above. We should just recommend using this first.

Reply:
Updated. PTAL